### PR TITLE
Parse envFile and pass to debugger

### LINKED ISSUE #1944: argsFile and envFile in launch.json
## Environment data
`dotnet --info` output:

```
.NET-Befehlszeilentools (2.0.2)
 
Product Information:
 Version:            2.0.2
 Commit SHA-1 hash:  a04b4bf512
 
Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.2\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d
```

VS Code version: 1.19.1
C# Extension version: 1.13.1

## Feature request
In my launch.json I need to pass sensitive data (like passwords) as arguments (or environment variables). As the launch.json is also committed to Git it's unwise to place the arguments in there. In NodeJS it is already possible to pass a file containing all the arguments (Microsoft/vscode#14523 and Microsoft/vscode-node-debug2@b3e2e8f3).

### PR DESCRIPTION
**Pull request by SebastianPfliegel**
Closes #1944. Adds a new option for launch.json - envFile: it will read environment variables from the given UTF8 file. This way environment variables can be passed to the debugger and launch.json stays clean of sensitive data (like passwords).

#### WIP
- [x] Remove UTF8-BOM if available
- [x] Ignore comments (using #)
- [ ] ~~Add similar logic for args~~

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
Looks like a good start to me

**Comment by gregg-miskelly** [comment]
@SebastianPfliegel is this ready for review or are you planning to add support for more things first?

**Comment by SebastianPfliegel** [comment]
It's still WIP. I'll try to finish it today or tomorrow.

**Comment by gregg-miskelly** [comment]
@SebastianPfliegel No rush on this, but just to let you know -- we are just about to start locking down for our next release (1.14). So if you get this done by early next week we can probably include it. Else, no worries, it can ship in a later release.

**Comment by heuristicAL** [comment]
Any updates on this PR? Since we're using Docker and a .env file at work, this would be a GREAT way for us to consolidate both configs as much as possible. Thanks! 😄

**Comment by SebastianPfliegel** [comment]
@heuristicAL, great to hear my PR is of use. I wasn't able to finish it as dotNet core was broken for my Linux distro, but it has been fixed last week, so I'll give it a try this week to finish it :)

**Comment by heuristicAL** [comment]
@SebastianPfliegel Great news! If you need any help let me know, I'm willing to give this a try, although my experience developing extensions for VS Code amounts to about.... 0. 👍

**Comment by SebastianPfliegel** [comment]
Haha, was the same for me. That's my first contribution regarding JavaScript/TypeScript ;)

**Comment by irq** [comment]
@SebastianPfliegel Eagerly awaiting this, any updates?

**Comment by heuristicAL** [comment]
Any updates on this? Would I be wasting my time if I were to fork out and update the PR?

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -100,11 +100,62 @@ export class CSharpConfigurationProvider implements vscode.DebugConfigurationPro
...
+
+                // Remove UTF-8 BOM if present
+                if(content.charAt(0) === '\uFEFF') {
+                    content = content.substr(1);
+                }
+
+                content.split("\n").forEach(line => {
+                    const r: RegExpMatchArray = line.match(/^\s*([\w\.\-]+)\s*=\s*(.*)?\s*$/);
+
+                    if (r !== null) {
```
[the remark this answers is not in the document]
Comment (gregg-miskelly) [reply src/configurationProvider.ts]: (Excluding lines that start with '#')

---
In reply to: [203184365](https://github.com/OmniSharp/omnisharp-vscode/pull/1951#discussion_r203184365) [](ancestors = 203184365)

File: package-lock.json [hunk 2]
```diff
@@ -6021,5130 +6021,6 @@
         "once": "1.4.0"
       }
     },
-    "npm": {
```
[the remark this answers is not in the document]
Comment (SebastianPfliegel) [reply package-lock.json]: Done!

File: src/configurationProvider.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (SebastianPfliegel) [reply src/configurationProvider.ts]: Also done!

**Comment by SebastianPfliegel** [comment]
Committed your requested changes. Should I squash them?

**Comment by gregg-miskelly** [comment]
Yes please.

**Comment by SebastianPfliegel** [comment]
Consider it done ;)

**Comment by SebastianPfliegel** [comment]
@gregg-miskelly: were you able to test it?

**Comment by gregg-miskelly** [comment]
@SebastianPfliegel Sorry for the delay. We had CI issues last week and I didn't have time to investigate. I will see if I can get this merged now...

**Comment by gregg-miskelly** [comment]
@akshita31: Would you mind merging this for me once the CI issues are fixed? I am going on vacation. If you need help from the debugger team chuckries is going to try and monitor things.

**Comment by akshita31** [comment]
@gregg-miskelly Sure, we are reverting to an older omnisharp version instead of the "latest" one for the integration tests. Will merge this, once we get the CI working.

**Comment by gregg-miskelly** [comment]
@SebastianPfliegel thanks so much for doing this! I made more changes to get this to work and created a new PR with both our work -- https://github.com/OmniSharp/omnisharp-vscode/pull/2462

I will close this one.

If you have the time, feel free to review my PR.

**Pull request closed without merging**